pkg/scheduler/event_handler.go

```diff
@@ -186,12 +190,69 @@ func (s *Scheduler) updateCluster(oldObj, newObj interface{}) {
 	case !equality.Semantic.DeepEqual(oldCluster.Labels, newCluster.Labels):
 		fallthrough
 	case oldCluster.Generation != newCluster.Generation:
-		s.enqueueAffectedBindings(oldCluster, newCluster)
+		s.clusterReconcileWorker.Add(oldCluster)
+		s.clusterReconcileWorker.Add(newCluster)
+	}
+}
+
+func (s *Scheduler) deleteCluster(obj interface{}) {
+	var cluster *clusterv1alpha1.Cluster
+	switch t := obj.(type) {
+	case *clusterv1alpha1.Cluster:
+		cluster = t
+	case cache.DeletedFinalStateUnknown:
+		var ok bool
+		cluster, ok = t.Obj.(*clusterv1alpha1.Cluster)
+		if !ok {
+			klog.Errorf("cannot convert to clusterv1alpha1.Cluster: %v", t.Obj)
+			return
+		}
+	default:
+		klog.Errorf("cannot convert to clusterv1alpha1.Cluster: %v", t)
+		return
+	}
+
+	klog.V(3).Infof("Delete event for cluster %s", cluster.Name)
+
+	if s.enableSchedulerEstimator {
+		s.schedulerEstimatorWorker.Add(cluster.Name)
+	}
+}
+
+func schedulerNameFilter(schedulerNameFromOptions, schedulerName string) bool {
+	if schedulerName == "" {
+		schedulerName = DefaultScheduler
+	}
+
+	return schedulerNameFromOptions == schedulerName
+}
+
+func (s *Scheduler) reconcileCluster(key util.QueueKey) error {
+	namespacedName, ok := key.(string)
+	if !ok {
+		return fmt.Errorf("failed to reconcileCluster as invalid key: %v", key)
```

Comment:
I think this is the root cause of the current CI failing, as the key is `not` a string.

Reply:
Please confirm it from the logs.

Reply:
From the log, it's true:
```log
worker.go:127] Dropping resource "member1" out of the queue: failed to reconcileCluster as invalid key: member1
```